agent/ecscni/plugin.go

```diff
@@ -187,13 +188,54 @@ func (client *cniClient) Version(name string) (string, error) {
 		return "", err
 	}
 
-	version := &struct {
-		Version string `json:"version"`
-	}{}
+	version := &cniPluginVersion{}
 	err = json.Unmarshal(versionInfo, version)
 	if err != nil {
 		return "", errors.Wrapf(err, "ecscni: Unmarshal version from string: %s", versionInfo)
 	}
 
-	return version.Version, nil
+	return version.str(), nil
+}
+
+// cniPluginVersion is used to convert the JSON output of the
+// '--version' command into a string
+type cniPluginVersion struct {
+	Version string `json:"version"`
+	Dirty   bool   `json:"dirty"`
+	Hash    string `json:"gitShortHash"`
+}
+
+func (version *cniPluginVersion) str() string {
+	ver := ""
+	if version.Dirty {
+		ver = "*"
+	}
+	return ver + version.Hash + "V" + version.Version
+}
+
+// Capabilities returns the capabilities supported by a plugin
+func (client *cniClient) Capabilities(name string) ([]string, error) {
+	file := filepath.Join(client.pluginsPath, name)
+
+	// Check if the plugin file exists before executing it
+	_, err := os.Stat(file)
+	if err != nil {
+		return nil, err
+	}
+
+	cmd := exec.Command(file, capabilitiesCommand)
+	capabilitiesInfo, err := cmd.Output()
```

Comment:
You may need to read from `Stderr` for the error message based on the implementation [here](https://github.com/aws/amazon-ecs-cni-plugins/blob/dev/plugins/ipam/main.go#L43).  Same to line 187.